libtenzir/builtins/formats/syslog.cpp

```diff
@@ -0,0 +1,108 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/argument_parser.hpp>
+#include <tenzir/format/reader_factory.hpp>
+#include <tenzir/format/syslog.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/table_slice_builder.hpp>
+#include <tenzir/to_lines.hpp>
+
+namespace tenzir::plugins::syslog {
+
+namespace {
+
+auto impl(generator<std::optional<std::string_view>> lines,
+          operator_control_plane& ctrl) -> generator<table_slice> {
+  for (auto&& line : lines) {
+    if (not line) {
+      co_yield {};
+      continue;
+    }
+    if (line->empty()) {
+      co_yield {};
+      continue;
+    }
+    const auto* f = line->begin();
+    const auto* const l = line->end();
+    std::optional<table_slice_builder> builder{};
+    format::syslog::message msg{};
+    format::syslog::legacy_message legacy_msg{};
+    if (auto parser = format::syslog::message_parser{}; parser(f, l, msg)) {
+      builder.emplace(format::syslog::make_syslog_type());
+      if (not builder->add(msg.hdr.facility, msg.hdr.severity, msg.hdr.version,
+                           msg.hdr.ts, msg.hdr.hostname, msg.hdr.app_name,
+                           msg.hdr.process_id, msg.hdr.msg_id, msg.msg)) {
+        diagnostic::error(
+          "syslog parser (RFC 5242) failed to produce table slice")
+          .hint(fmt::format("line: `{}`", *line))
+          .emit(ctrl.diagnostics());
+        co_return;
+      }
+    } else if (auto legacy_parser = format::syslog::legacy_message_parser{};
+               legacy_parser(f, l, legacy_msg)) {
+      builder.emplace(format::syslog::make_legacy_syslog_type());
+      if (not builder->add(legacy_msg.facility, legacy_msg.severity,
+                           legacy_msg.timestamp, legacy_msg.host,
+                           legacy_msg.tag, legacy_msg.content)) {
+        diagnostic::error(
+          "syslog parser (RFC 3164) failed to produce table slice")
+          .hint(fmt::format("line: `{}`", *line))
+          .emit(ctrl.diagnostics());
+        co_return;
+      }
+    } else {
+      builder.emplace(format::syslog::make_unknown_type());
+      if (not builder->add(*line)) {
+        diagnostic::error(
+          "syslog parser (unknown format) failed to produce table slice")
+          .hint(fmt::format("line: `{}`", *line))
+          .emit(ctrl.diagnostics());
+        co_return;
+      }
```

Comment:
I can see why this might be useful. In the JSON parser, we took a different approach and discard unparseable events instead, but emit a warning. The problem is that most downstream operators will not expect this format. But perhaps that's okay until we figure out a proper story for handling malformed input?

Reply:
I think this is fine for now. It's hard to decipher whether something is a oddly-structured syslog message, or something that really isn't a syslog message at all. Since it seems that those former cases are quite prevalent, it'd be a shame to throw those in the trash.